WebInstall/explainer.md

```diff
@@ -1,10 +1,363 @@
+
+This results in a new integration with the [Permissions API](https://www.w3.org/TR/permissions/). The install API will make available the "installation" [PermissionDescriptor](https://www.w3.org/TR/permissions/#dom-permissiondescriptor) as a new [*powerful feature*](https://www.w3.org/TR/permissions/#dfn-specifies-a-powerful-feature). This would make it possible to know programmatically if `install` would be blocked.
+
+```javascript
+/* example of querying for the state of an installation permission using the Permission API  */
+
+const { state } = await navigator.permissions.query({
+  name: "installation"
+});
+switch (state) {
+  case "granted":
+    navigator.install('https://productivitysuite.com');
+    break;
+  case "prompt":
+    //shows the install button in the web UI
+    showInstallButton();
+    break;
+  case "denied":
+    redirectToApp();
+    break;
+}
+```
+
+> **Note:** A permission prompt will appear for origins that do not have the capability to install apps if the invoking document is not the current one. Even if the installation is of a "[background document](#background-document-1-param)" in the same origin, for consistency the origin must have the permission to install apps. The only cases that will not prompt for the permission are the installation of the "[current document](#current-document)" or a "[background document](#background-document-1-param)" in an origin that already has installation permissions.
+
+**Example:**
+
+The app located in `https://productivitysuite.com` displays in its homepage 3 buttons that aim to install 3 different apps (notice all apps are in the same origin):
+* the text processor located at `https://productivitysuite.com/text`
+* the presentation app located at `https://productivitysuite.com/slides`
+* the spreadsheet locacated at `https://productivitysuite.com/spreadsheet`
+
+The end user goes to the homepage in the `https://productivitysuite.com`'s origin and clicks on the button to install the presentation application. As this is a *background document* (not the current document the user is interacting with) and the origin does not have permission to install apps, a permission prompt will appear. If this permission is granted for the origin, it can now install apps. After this permissin prompt the second prompt where the user confirms the isntallation appears.  
+
+The end user then tries to install the text processor, and since the origin has been granted the permission, then it will prompt to confirm that "productivity suite wants to install text procesor". The installation permission is bound to an origin.
+
+If the user where to deny the permission to install for the origin, they could browse to the app itself and once there, they can install the application. In this case there isn't any permission prompt required as this is now a *current document* installation. 
+
+
+### Rejecting promise with `AbortError` and `DataError`
```

Comment:
You should explicitly call out the scenario Dan mentioned that talked about how this could have been exploited by a bad actor (to determine if a user was logged-in to a site that had the manifest behind a login).

Reply:
Was this a Won't Fix to the comment, or did you forget to do it?

Reply:
I added an open question, since the topic that started the feedback was the id mismatch error we were thinking on implementing earlier in the API design. Do you recon this needs more information?

Reply:
I've checked on the conversation in https://docs.google.com/document/d/1RNYtHTi9N6PsYcjzSWzy9Zu3H4qwzraI1KcItVA2rIE/edit?tab=t.0#heading=h.o94qpooinv0p . Per conversation the use of error grouping does mitigate these use cases where a manifest would be served behind a log-in.